app/cleanup_manager.py

```
@@ -0,0 +1,27 @@
+from __future__ import annotations
+from typing import Type
+import os
+from traceback import TracebackException
+
+
+class CleanUpFile:
+    def __init__(self, filename: str) -> None:
+        self.filename = filename
+
+    def __enter__(self) -> CleanUpFile:
+        if not os.path.exists(self.filename):
+            raise ValueError("файлу нема")
+        return self
+
+    def __exit__(
+            self,
+            exc_type: Type[BaseException],
+            exc_val: BaseException,
+            exc_tb: TracebackException
+    ) -> bool:
+        os.remove(self.filename)
+        return True
```

This implementation can cause data loss. The `__exit__` method is called even when an exception occurs. By unconditionally removing the file and returning `True`, you delete the source file even if the copy failed, and you hide the original error. The file should only be removed if no exception occurred (when `exc_type` is `None`), and exceptions should be allowed to propagate by returning `False` or `None`.